pkg/discovery/discovery.go

```diff
@@ -80,6 +81,17 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) []error {
 		errlst = append(errlst, err)
 	}
 
+	// 8. Always write the done file
+	doneFile := cfg.ResultsDir + "/done"
```

Comment:
> This change allows us to get both the blocking behavior as well as a done file. 

Under what conditions do we want blocking + done?  
done indicates hand-off to another process in the pod.  

I'm not opposed to the logic of adding to the process, just wanted to make sure we vetted.  But I do think that done && --no-exit are an mutually exclusive.  If you specify done you are stating that you don't want to block, but instead you want to exit.  

Perhaps we should call it something other then RESULTS_DIR, to make it more explicit.